### ISSUE TITLE
docker makefile - error on run

### ISSUE THREAD

**Issue by gerbrent** [JupiterBroadcasting/jupiterbroadcasting.com#217]
I get the following when running `make dev`

```
brent@x250 ~/c/j/jupiterbroadcasting-com (main)> make dev

if [ "hugo" == hugo ] ; then \
        hugo version | grep 'extended' >/dev/null || (echo "You don't have the extended version of hugo, please install it" && exit 1); \
else \
        docker build --rm -f Dockerfile -t jb_hugo:latest . ;\
fi
hugo serve -D
Start building sites … 
hugo v0.93.3+extended linux/amd64 BuildDate=unknown

```

this if statement is a direct cat of what's in the `Makefile`. Doesn't seem like the desired behaviour....

likely related to changes performed in #170

**Comment by elreydetoda** [comment]
Ya...that's interesting, it's the first time I've made a Makefile, so I'll investigate this after #110.

Did it still work properly? Like did hugo start serving the pages?

**Comment by gerbrent** [comment]
On one system (known good hugo dev system) - yes.
On another (new-ish KDE NEON install; docker, hugo installed.) - no:

```
brent@NEON ~/j/h/code (main) [2]> make
if [ "hugo" == hugo ] ; then \
        hugo version | grep 'extended' >/dev/null || (echo "You don't have the extended version of hugo, please install it" && exit 1); \
else \
        docker build --rm -f Dockerfile -t jb_hugo:latest . ;\
fi
hugo serve -D
Error: "/home/alex/jb/hugo/code/config/_default/params.toml:6:1": unmarshal failed: Near line 6 (last key parsed ''): bare keys cannot contain '.'
make: *** [Makefile:16: dev] Error 255
```

**Comment by elreydetoda** [comment]
Interesting...it worked perfectly fine on my OS with docker...I guess we're on different versions or something, alright I'll see what I can do with that...😞😅

**Comment by gerbrent** [comment]
let me know if you need help/input!

Afterall, isn't the whole point of a docker dev workflow to be system agnostic? (Nix, where are you?! ; )

**Comment by CGBassPlayer** [comment]
I'll throw in that it does work for me on my system
* Pop OS 22.04
  * Docker 20.10.14
  * Make 4.3

**Comment by elreydetoda** [comment]
This'll depend on if #239 will be merged or not, because the only dev model will be via docker.

Also, @gerbrent it looks like for this:


> ```
> hugo serve -D
> Error: "/home/alex/jb/hugo/code/config/_default/params.toml:6:1": unmarshal failed: Near line 6 (last key parsed ''): bare keys cannot contain '.'
> make: *** [Makefile:16: dev] Error 255
> ```


It looks like there might have been something wrong with your hugo config. It also looks like it's using a hugo version that was installed on the system (based on it saying the `hugo serve` command. One thing I did was check for the hugo binary first, and use that instead of a container.

That was to help prevent the overhead of spinning up a container if you'd already installed hugo. Based on #239 though, it'll force docker development only.

**Comment by gerbrent** [comment]
thanks for the info @elreydetoda !

I think by #239 surely you mean another... perhaps #249 ?

And agree - if #249 is indeed merged, then this issue goes away aswell.... bonus!

**Comment by elreydetoda** [comment]
lol, ya that's what I meant :sweat_smile: . (I was on mobile at the time and they don't give you suggestions in the GH app :sweat_smile: )

**Comment by gerbrent** [comment]
(I will edit your mis-tagged issue above, just to make it easier to read for the future visitors)

**Comment by gerbrent** [comment]
solved by [#4864df857e4eee03064e4077271b906ebecc7095](https://github.com/JupiterBroadcasting/jupiterbroadcasting.com/pull/249/commits/4864df857e4eee03064e4077271b906ebecc7095)
